In [1]:
! git clone https://github.com/mohan522/IPL.git


Cloning into 'IPL'...
remote: Enumerating objects: 40, done.
remote: Counting objects: 100% (40/40), done.
remote: Compressing objects: 100% (32/32), done.
remote: Total 40 (delta 10), reused 16 (delta 3), pack-reused 0
Unpacking objects: 100% (40/40), done.


In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


In [3]:
ball_df = pd.read_csv("/content/IPL/Ball_By_Ball.csv")
match_df = pd.read_csv("/content/IPL/Match.csv")


In [5]:
player_df = pd.read_csv("/content/IPL/Player.csv",engine = "python")
plyr_mtch_df = pd.read_csv("/content/IPL/Player_match.csv",engine = "python")
team_df = pd.read_csv("/content/IPL/Team.csv",engine = "python")

In [6]:
player_df.head()

PLAYER_SK  Player_Id  ...       Bowling_skill Country_Name
0          0          1  ...    Right-arm medium        India
1          1          2  ...    Right-arm medium  New Zealand
2          2          3  ...    Right-arm medium    Australia
3          3          4  ...  Right-arm offbreak    Australia
4          4          5  ...  Right-arm offbreak     Pakistan

[5 rows x 7 columns]

In [7]:
plyr_mtch_df.head()

Player_match_SK  PlayerMatch_key  ...  Player_keeper  Opposit_keeper
0               -1    -1.000000e+00  ...            NaN             NaN
1            12694     3.359870e+10  ...     MV Boucher         WP Saha
2            12695     3.359870e+10  ...     MV Boucher         WP Saha
3            12696     3.359870e+10  ...     MV Boucher         WP Saha
4            12697     3.359870e+10  ...     MV Boucher         WP Saha

[5 rows x 22 columns]

In [8]:
match_df.head()

Match_SK  match_id  ... Win_Margin Country_id
0         0    335987  ...      140.0          1
1         1    335988  ...       33.0          1
2         2    335989  ...        9.0          1
3         3    335990  ...        5.0          1
4         4    335991  ...        5.0          1

[5 rows x 17 columns]

In [10]:
match_df.describe()

Match_SK      match_id  Season_Year  Win_Margin  Country_id
count  637.000000  6.370000e+02   637.000000  628.000000  637.000000
mean   318.000000  6.378825e+05  2012.497645   17.272293    1.152276
std    184.030342  2.356312e+05     2.776600   22.212632    0.438388
min      0.000000  3.359870e+05  2008.000000    0.000000    1.000000
25%    159.000000  4.191550e+05  2010.000000    6.000000    1.000000
50%    318.000000  5.483830e+05  2012.000000    8.000000    1.000000
75%    477.000000  8.297480e+05  2015.000000   20.000000    1.000000
max    636.000000  1.082650e+06  2017.000000  146.000000    3.000000

In [11]:
match_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 637 entries, 0 to 636
Data columns (total 17 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Match_SK      637 non-null    int64  
 1   match_id      637 non-null    int64  
 2   Team1         637 non-null    object 
 3   Team2         637 non-null    object 
 4   match_date    637 non-null    object 
 5   Season_Year   637 non-null    int64  
 6   Venue_Name    636 non-null    object 
 7   City_Name     637 non-null    object 
 8   Country_Name  637 non-null    object 
 9   Toss_Winner   636 non-null    object 
 10  match_winner  634 non-null    object 
 11  Toss_Name     636 non-null    object 
 12  Win_Type      635 non-null    object 
 13  Outcome_Type  637 non-null    object 
 14  ManOfMach     633 non-null    object 
 15  Win_Margin    628 non-null    float64
 16  Country_id    637 non-null    int64  
dtypes: float64(1), int64(4), object(12)
memory usage: 84.7+ KB


##From the above info one can say that there are 637 macthes in total
##From below info matches are played from 2008-2017

In [12]:
#matches Period
match_df.Season_Year.unique()

array([2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017])

#Questions to answer from the above data are...


1.   Toss winning is a factor in winning the match?
2.   What Venue and city hosted most of the matches in each series?
3.   Having a Specific Captain makes any difference in wining a Match?
4.   Teams total wins per each season and entire period?
5.   which Team won most when toss is against them per season




In [29]:
match_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 637 entries, 0 to 636
Data columns (total 17 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Match_SK      637 non-null    int64  
 1   match_id      637 non-null    int64  
 2   Team1         637 non-null    object 
 3   Team2         637 non-null    object 
 4   match_date    637 non-null    object 
 5   Season_Year   637 non-null    int64  
 6   Venue_Name    636 non-null    object 
 7   City_Name     637 non-null    object 
 8   Country_Name  637 non-null    object 
 9   Toss_Winner   636 non-null    object 
 10  match_winner  634 non-null    object 
 11  Toss_Name     636 non-null    object 
 12  Win_Type      635 non-null    object 
 13  Outcome_Type  637 non-null    object 
 14  ManOfMach     633 non-null    object 
 15  Win_Margin    628 non-null    float64
 16  Country_id    637 non-null    int64  
dtypes: float64(1), int64(4), object(12)
memory usage: 84.7+ KB


In [13]:
match_df.Outcome_Type.unique()

array(['Result', 'Superover', 'No Result', 'abandoned', 'tied'],
      dtype=object)

In [14]:
match_df.Outcome_Type.value_counts()

Result       626
Superover      6
No Result      3
abandoned      1
tied           1
Name: Outcome_Type, dtype: int64

In [15]:
for i in range(0,len(match_df.Outcome_Type)):
  if match_df.Outcome_Type[i] == "Result":
    match_df["winning"] = 1
  elif match_df.Outcome_Type[i] == "Superover":
    match_df["winning"] = 2
  elif match_df.Outcome_Type[i] == "No Result":
    match_df["winning"] = 3
  elif match_df.Outcome_Type[i] == "tied":
    match_df["winning"] = 4
  elif match_df.Outcome_Type[i] == "abandoned":
    match_df["winning"] = 5
  i=i+1


In [16]:
match_df["winning"][636]

1

In [17]:
match_df.Toss_Winner

0      Royal Challengers Bangalore
1              Chennai Super Kings
2                 Rajasthan Royals
3                   Mumbai Indians
4                  Deccan Chargers
                  ...             
632    Royal Challengers Bangalore
633                 Mumbai Indians
634          Kolkata Knight Riders
635                 Mumbai Indians
636                 Mumbai Indians
Name: Toss_Winner, Length: 637, dtype: object

In [18]:
a = 0
for i in range(0,len(match_df.Toss_Winner)):
  if match_df.Toss_Winner[i] == match_df.Team1[i]:
    a = a+1
print(a)

307


In [19]:
a/match_df.Toss_Winner.count()

0.4827044025157233

## The above code proves that Winning Toss is not a strong factor for winning the game!

In [20]:
match_df.head()

Match_SK  match_id  ... Country_id winning
0         0    335987  ...          1       1
1         1    335988  ...          1       1
2         2    335989  ...          1       1
3         3    335990  ...          1       1
4         4    335991  ...          1       1

[5 rows x 18 columns]

In [37]:
 match_df.groupby("Venue_Name").count().max()



Match_SK        66
match_id        66
Team1           66
Team2           66
match_date      66
Season_Year     66
City_Name       66
Country_Name    66
Toss_Winner     66
match_winner    64
Toss_Name       66
Win_Type        66
Outcome_Type    66
ManOfMach       64
Win_Margin      63
Country_id      66
winning         66
dtype: int64

In [35]:
print(grouped_df.agg())

AssertionError: ignored

array(['M Chinnaswamy Stadium',
       'Punjab Cricket Association Stadium, Mohali', 'Feroz Shah Kotla',
       'Wankhede Stadium', 'Eden Gardens', 'Sawai Mansingh Stadium',
       'Rajiv Gandhi International Stadium, Uppal',
       'MA Chidambaram Stadium, Chepauk', 'Dr DY Patil Sports Academy',
       'Newlands', "St George's Park", 'Kingsmead', 'SuperSport Park',
       'Buffalo Park', 'New Wanderers Stadium', 'De Beers Diamond Oval',
       'OUTsurance Oval', 'Brabourne Stadium',
       'Sardar Patel Stadium, Motera', 'Barabati Stadium',
       'Vidarbha Cricket Association Stadium, Jamtha',
       'Himachal Pradesh Cricket Association Stadium', 'Nehru Stadium',
       'Holkar Cricket Stadium',
       'Dr. Y.S. Rajasekhara Reddy ACA-VDCA Cricket Stadium',
       'Subrata Roy Sahara Stadium',
       'Shaheed Veer Narayan Singh International Stadium',
       'JSCA International Stadium Complex', 'Sheikh Zayed Stadium',
       'Sharjah Cricket Stadium', 'Dubai International Cricket St